plugins/websocket_server/WebsocketServer.cc

```diff
@@ -998,6 +998,40 @@ void WebsocketServer::OnMessage(int _socketId, const std::string _msg)
   {
     this->OnAsset(_socketId, frameParts);
   }
+  else if (frameParts[0] == this->operations[SERVICE])
+  {
+    this->OnService(_socketId, frameParts);
+  }
+}
+
+//////////////////////////////////////////////////
+void WebsocketServer::OnService(int _socketId,
+    const std::vector<std::string> &_frameParts)
+{
+  std::string service = _frameParts[1];
+  std::string msgTypeName = _frameParts[2];
+  std::string msgData = _frameParts[3];
+
+  igndbg << "Calling service [" << service << "]\n";
+  bool result;
+  unsigned int timeout = 2000;
+
+  std::string repStr;
+  std::string repTypeName;
+  bool executed = this->node.RequestRaw(service, msgData, msgTypeName,
+                                        timeout, repStr, repTypeName, result);
+  if (!executed)
+  {
+    ignerr << "Unable to call service [" << service  << "]\n";
+  }
+
+  // Construct the response message
+  std::string data = BUILD_MSG(this->operations[SERVICE], service,
```

Comment:
I'm not 100% sure how necessary is to have a response here, but I guess it wouldn't hurt. Future services may benefit from this.